src/gui/mzroll/mainwindow.cpp

```diff
@@ -892,26 +880,36 @@ void MainWindow::_setProjectFilenameFromProjectDockWidget()
     auto lastLoad = projectDockWidget->getLastOpenedTime();
     if (!projectDockWidget->getLastOpenedProject().isEmpty()
             && lastLoad > lastSave)
-        fileName = projectDockWidget->getLastOpenedProject();
+        _currentProjectName = projectDockWidget->getLastOpenedProject();
     if (!projectDockWidget->getLastSavedProject().isEmpty()
             && lastSave > lastLoad)
-        fileName = projectDockWidget->getLastSavedProject();
+        _currentProjectName = projectDockWidget->getLastSavedProject();
+}
+
+void MainWindow::resetAutosave()
+{
+    if (this->autosaveEnabled)
+        QFile::remove(_currentProjectName);
+    this->autosaveEnabled = false;
+    this->peaksMarked = 0;
+    _currentProjectName = "";
 }
 
 void MainWindow::autosaveProject()
 {
-    if (this->peaksMarked == 1 && this->askAutosaveMain == 0) {
-        this->askAutosaveMain++;
-        this->doAutosave = this->askAutosave();
-        if (this->doAutosave) {
-            _setProjectFilenameIfEmpty();
-            if (this->fileName.isEmpty()) {
-                this->doAutosave = false;
-                return;
-            }
-            autosave->saveProjectWorker();
+    if (this->peaksMarked == 1) {
+        if (!this->autosaveEnabled)
```

Comment:
'autosaveEnabled': Please change the variable name to something more explanatory.